src/pages/game.js

```diff
@@ -33,141 +31,181 @@ export default function Game({data, location}) {
+
+   const usersTurn = (cards = {}) => {
+      setAllState({...allState, ...cards, gameState: GAME_STATE_YOUR_TURN})
+   }
 
-   const incrementTurnCount = () => setTurnCount(turnCount + 1)
+   const computersTurn = (cards = {}) => {
 
-   const startGame = () => {
+      const {
+         name,
+         cardDescription,
+         mugshot,
+         mugshotAltText,
+         ...attributes
+      } = allState.computersCards[0]
+      const attributesArray = map(attributes, (value, key) => ({key: key, value: value}))
+      const orderedAttributes = orderBy(attributesArray, ["value"], ["desc"])
 
-      setComputerName(shuffle(names)[0])
-      const shuffledCards = shuffle(cards)
-      const splitCards = chunk(shuffledCards, shuffledCards.length / 2)
+      setAllState({...allState, ...cards, gameState: GAME_STATE_COMPUTER_TURN})
 
-      setUsersCards(splitCards[0])
-      setComputersCards(splitCards[1])
+      setTimeout(() => slamJams(orderedAttributes[0].key), 1500)
+   }
 
-      incrementTurnCount()
+   const takeTurn = attribute => {
+      setAllState({...allState, gameState: GAME_STATE_WAITING_FOR_COMPUTER})
+      setTimeout(() => slamJams(attribute), 700);
    }
 
    const drawCard = () => {
-      setComputersTurnCard(computersCards[0])
-      setUsersTurnCard(usersCards[0])
 
-      setUsersCards(usersCards.slice(1))
-      setComputersCards(computersCards.slice(1))
-   }
+      if (allState.roundWinner === PLAYER_USER) {
+         return {
+            computersCards: allState.computersCards.splice(1),
+            usersCards: [...allState.usersCards.splice(1), allState.usersCards[0], allState.computersCards[0]],
+         }
+      }
 
-   const computersTurn = () => {
-      const {name, cardDescription, mugshot, mugshotAltText, ...attributes} = computersTurnCard
+      if (allState.roundWinner === PLAYER_COMPUTER) {
+         return {
+            usersCards: allState.usersCards.splice(1),
+            computersCards: [...allState.computersCards.splice(1), allState.computersCards[0], allState.usersCards[0]],
+         }
+      }
 
-      const attributesArray = map(attributes, (value, key) => ({key: key, value: value}))
-      console.log(attributesArray);
-      const orderedAttributes = orderBy(attributesArray, ["value"], ["desc"])
-      setTimeout(() => slamJams(orderedAttributes[0].key), 1500)
+      return {
+         usersCards: [...allState.usersCards.splice(1), allState.usersCards[0]],
+         computersCards: [...allState.computersCards.splice(1), allState.computersCards[0]]
+      }
    }
 
    const slamJams = attribute => {
-      setSelectedAttribute(attribute)
 
-      const hasUserWon = usersTurnCard[attribute] > computersTurnCard[attribute]
-      const isDraw = usersTurnCard[attribute] === computersTurnCard[attribute]
+      const hasUserWon = allState.usersCards[0][attribute] > allState.computersCards[0][attribute]
```

Comment:
I know we've removed the singular user/computer card from state, but it might be helpful to have some local variables, just inside these functions? You can also use array restructuring like so 
```js
> const [head, ...tail] = ['x', 'y', 'z']
> head
'x'
> tail
[ 'y', 'z' ]
```

which might be quite useful, e.g.
```js
const [usersCard, ...usersRemaining] = allState.usersCards;
```